lib/widgets/new_dm_sheet.dart

```diff
@@ -0,0 +1,327 @@
+      children: [
+        Icon(ZulipIcons.chevron_left, size: 24, color: designVariables.icon),
+        Text(zulipLocalizations.newDmSheetBackButtonLabel,
+          style: TextStyle(fontSize: 20, color: designVariables.icon))]));
+
+  Widget _buildNextButton(BuildContext context, PerAccountStore store,
+      DesignVariables designVariables, ZulipLocalizations zulipLocalizations) => TextButton(
+    onPressed: selectedUserIds.isEmpty ? null : () {
+      final narrow = DmNarrow.withOtherUsers(
+        selectedUserIds,
+        selfUserId: store.selfUserId);
+      Navigator.pop(context);
+      Navigator.push(context, MessageListPage.buildRoute(
+        context: context,
+        narrow: narrow));
+    },
+    child: Row(
+      mainAxisSize: MainAxisSize.min,
+      children: [
+        Text(zulipLocalizations.newDmSheetNextButtonLabel,
+          style: TextStyle(
+            fontSize: 20,
+            fontWeight: FontWeight.w600,
+            color: selectedUserIds.isEmpty
+              ? designVariables.icon.withFadedAlpha(0.5)
+              : designVariables.icon)),
+        Icon(ZulipIcons.chevron_right,
+          size: 24,
+          color: selectedUserIds.isEmpty
+            ? designVariables.icon.withFadedAlpha(0.5)
+            : designVariables.icon)]));
+}
+
+class NewDmSearchBar extends StatelessWidget {
+  const NewDmSearchBar({
+    super.key,
+    required this.searchController,
+    required this.scrollController,
+    required this.selectedUserIds});
+
+  final TextEditingController searchController;
+  final ScrollController scrollController;
+  final Set<int> selectedUserIds;
+
+  @override
+  Widget build(BuildContext context) {
+    final designVariables = DesignVariables.of(context);
+    final zulipLocalizations = ZulipLocalizations.of(context);
+
+    return Container(
+      width: MediaQuery.of(context).size.width,
+      constraints: const BoxConstraints(
+        minHeight: 44,
+        maxHeight: 124),
+      padding: const EdgeInsets.symmetric(horizontal: 14,vertical:11),
+      decoration: BoxDecoration(color: designVariables.bgSearchInput),
+      child: SingleChildScrollView(
+        controller: scrollController,
+        child: Row(children: [
+          Expanded(child: Wrap(
+            spacing: 6,
+            runSpacing: 4,
+            crossAxisAlignment: WrapCrossAlignment.center,
+            children: [
+              ...selectedUserIds.map((userId) => SelectedUserChip(userId: userId)),
+              _buildSearchField(designVariables, zulipLocalizations)]))])));
+  }
+
+  Widget _buildSearchField(DesignVariables designVariables,
+      ZulipLocalizations zulipLocalizations) => IntrinsicWidth(
+        child: SizedBox(
+          height: 22,
+          child: TextField(
+            controller: searchController,
+            cursorColor: designVariables.foreground,
+            style: const TextStyle(
+              fontSize: 17,
+              height: 1.0,
```

Comment:
In the design, the line-height is 22px:

![image](https://github.com/user-attachments/assets/bb1e3c73-5971-4db1-9e72-00eea523af6c)

See my other comment on `height`. We usually express them as fractions.